modules/ROOT/pages/kafka/kafka-connector-upgrade-migrate.adoc

```diff
@@ -0,0 +1,131 @@
+
+|Describe | Retrieves the subscriptions and assignments of the associated connection, along with the current offsets for each topic-partition. This operation blocks all the consumers, so there are no changes to the offsets assignments or other information that is retrieved by this operation. |
+
+| Publish | The `publish` operation enables sending messages to a Kafka topic, and is non-blocking. Its behavior depends on specific values of the connection parameters. This operation supports transactions, which means that a transaction ID is randomly generated for the Kafka producer and handled in the connection.
+a| *Topic (The topic name)
+* Partition (The partition name)
+* Key
+* Message
+* Headers
+| Seek | Sets the current offset of the consumer for the given topic and partition to the provided offset value. a| * Topic (The topic name)
+* Partition (The partition name)
+* Offset (The offset to seek to) |
+|===
+
+[[new_sources]]
+== New Sources
+
+[%header%autowidth.spread]
+|===
+|Kafka Operation | Description | Parameters
+
+| Batch message listener
+a| Works in the same way as the message listener except that it processes the list of messages instead of processing a single message at a time. The message list that was obtained in the poll is handled by a flow as a single event, so the handling of concurrency is simpler than in the simple message listener. In other words, a commit of the messages happens for all the messages as a whole by calling the commit on Kafka consumer.
+a| * Poll timeout
+* Poll timeout time unit
+* Acknowledgment mode
+* Number of parallel consumers
+|===
+
+
+== Upgrade Prerequisites
+
+Before you perform the upgrade, you must create a backup of your files, data, and configurations in case you need to revert to the previous version.
+
+== Upgrade Steps
+
+Follow these steps to perform the upgrade to Kafka Connector 4.0.x:
+
+. In Studio, create a Mule project.
+. In the Mule Palette view, click *Search in Exchange*.
+. In Add Modules to Project, type 'Kafka' in the search field.
+. In Available modules, select *Kafka Connector*, and then click *Add*.
+. Click *Finish*. +
+Anypoint Studio upgrades the connector automatically.
+. Verify that the `kafka-connector` dependency version is `4.0.x` in the pom.xml.
+
+
+== Post Upgrade Steps
+
+After you install the latest version of the connector, follow these steps to complete the upgrade:
+
+. In Anypoint Studio, verify that there are no errors in the *Problems* or *Console* views.
+. Check the project pom.xml and verify that there are no problems.
+. Test the connection and verify that the operations work.
+
+== Troubleshooting
+
+If there are problems with caching the parameters and caching the metadata, try restarting Anypoint Studio.
+
+=== Reverting the Upgrade
+
+If it is necessary to revert to the previous version of Kafka Connector, in Anypoint Studio, change the `kafka-connector` dependency version `4.0.x` in the project's pom.xml to the previous version.
```

Comment:
ditto
Diane formats this differently as well. You two should prolly coordinate at some point. I know you are in a rush right now.